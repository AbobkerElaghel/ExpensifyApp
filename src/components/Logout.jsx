import React from "react";
import {startLogout} from "../actions/authAction";
import {connect} from "react-redux";

export const Logout = ({startLogout}) => (
    <div>
        <button onClick={startLogout}>Logout</button>
    </div>
);

const mapDispatchToProps = dispatch => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Logout);