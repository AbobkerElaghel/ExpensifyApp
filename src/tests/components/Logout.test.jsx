import React from "react";
import { shallow } from "enzyme";
import {Logout} from '../../components/Logout.jsx'

test("should render Logout component correctly", () => {
    const wrapper = shallow(<Logout />);
    expect(wrapper).toMatchSnapshot();
});

test("should call startLogout action", () => {
    const startLogoutSpy = jest.fn();
    const wrapper = shallow(<Logout startLogout={startLogoutSpy}/>);
    wrapper.find('button').simulate('click');
    expect(startLogoutSpy).toBeCalled();
});