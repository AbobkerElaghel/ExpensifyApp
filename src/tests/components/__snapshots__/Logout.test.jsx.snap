// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Logout component correctly 1`] = `
<div>
  <button>
    Logout
  </button>
</div>
`;
